jsettlers.graphics/src/jsettlers/graphics/map/controls/original/panel/content/BarFill.java

```diff
@@ -0,0 +1,89 @@
+/*******************************************************************************
+ * Copyright (c) 2015
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *******************************************************************************/
+package jsettlers.graphics.map.controls.original.panel.content;
+
+import go.graphics.GLDrawContext;
+import jsettlers.common.images.EImageLinkType;
+import jsettlers.common.images.ImageLink;
+import jsettlers.common.images.OriginalImageLink;
+import jsettlers.common.position.FloatRectangle;
+import jsettlers.graphics.action.Action;
+import jsettlers.graphics.action.ExecutableAction;
+import jsettlers.graphics.utils.UIPanel;
+
+public class BarFill extends UIPanel
+{
+    private static final ImageLink barImageLink = new OriginalImageLink(EImageLinkType.GUI, 3, 330, 0); //or 336
+
+    private final UIPanel fill;
+    private final UIPanel frame;
+    private final ExecutableAction action;
+    private ExecutableAction listener;
+    private float value;
+
+    public BarFill()
+    {
+        fill = new UIPanel(){
+            @Override
+            public void drawAt(GLDrawContext gl) {
+                gl.color(0f, 0.78f, 0.78f, 1f);
+                FloatRectangle position = getPosition();
+                gl.fillQuad(position.getMinX(), position.getMinY(), position.getMaxX(), position.getMaxY());
+            }
+        };
+
+        frame = new UIPanel();
+        frame.setBackground(barImageLink);
+        addChild(frame, 0f, 0f, 1f, 1f);
+
+        action = new ExecutableAction() {
+            @Override
+            public void execute() {
+                setBarFill(value);
+                if(listener != null){
+                    listener.execute();
+                }
+            }
+        };
+    }
+
+    @Override
+    public Action getAction(float relativex, float relativey)
+    {
+        value = relativex;
```

Comment:
This does not need to be caused by a mouse event. You should not set value here.